com.unity.render-pipelines.core/Editor/SpeedTree8MaterialUpgrader.cs

```diff
@@ -0,0 +1,181 @@
+            }
+            if (mat.HasProperty("_CullMode"))
+                mat.SetFloat("_CullMode", cullmode);
+        }
+
+        /// <summary>
+        /// Postprocess materials when importing a SpeedTree8 asset. Call from OnPostprocessSpeedTree in a MaterialPostprocessor.
+        /// </summary>
+        /// <param name="speedtree">Game object for the SpeedTree asset being imported.</param>
+        /// <param name="stImporter">The assetimporter used to import the SpeedTree asset.</param>
+        /// <param name="finalizer">Render pipeline-specific material finalizer.</param>
+        public static void PostprocessSpeedTree8Materials(GameObject speedtree, SpeedTreeImporter stImporter, MaterialFinalizer finalizer = null)
+        {
+            LODGroup lg = speedtree.GetComponent<LODGroup>();
+            LOD[] lods = lg.GetLODs();
+            for (int l = 0; l < lods.Length; l++)
+            {
+                LOD lod = lods[l];
+                bool isBillboard = stImporter.hasBillboard && (l == lods.Length - 1);
+                int wq = Mathf.Min(stImporter.windQualities[l], stImporter.bestWindQuality);
+                foreach (Renderer r in lod.renderers)
+                {
+                    foreach (Material m in r.sharedMaterials)
+                    {
+                        ImportNewSpeedTree8Material(m, wq, isBillboard);
+                        if (finalizer != null)
+                            finalizer(m);
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Preserve wind quality and billboard setting when upgrading a ST8 material from previous versions.
+        /// Wind priority order is enabled keyword > _WindQuality float value.
+        /// Should work for upgrading versions within a pipeline and from standard to current pipeline.
+        /// </summary>
+        /// <param name="material">SpeedTree8 material to upgrade.</param>
+        public static void SpeedTree8MaterialFinalizer(Material material)
+        {
+            if (material.HasProperty("_TwoSided") && material.HasProperty("_CullMode"))
+                material.SetFloat("_CullMode", material.GetFloat("_TwoSided"));
+
+            if (material.IsKeywordEnabled("EFFECT_EXTRA_TEX"))
+                material.SetFloat("EFFECT_EXTRA_TEX", 1.0f);
+
+            bool isBillboard = material.IsKeywordEnabled("EFFECT_BILLBOARD");
+            if (material.HasProperty("EFFECT_BILLBOARD"))
+                material.SetFloat("EFFECT_BILLBOARD", isBillboard ? 1.0f : 0.0f);
+
+            UpgradeWindQuality(material);
+        }
+
+        private static void UpgradeWindQuality(Material material, int windQuality = -1)
+        {
+            int wq = GetWindQuality(material, windQuality);
+            SetWindQuality(material, wq);
+        }
+
+        private static int GetWindQuality(Material material, int windQuality = -1)
+        {
+            // Conservative wind quality priority:
```

Comment:
Does quality priority relate to a failover from higher to lower quality?
Does "conservative" mean more likely to failover, or less likely to failover? (Or something else, if this doesn't have to do with failover)